### PR TITLE
docs: document Route 53 and S3+DynamoDB permissions

### PR DESCRIPTION
**Pull request by iaguis**
For the Route 53 DNS provider and the S3 configuration backend.

I didn't touch the quickstarts yet since they're being reworked.

Fixes https://github.com/kinvolk/lokomotive/issues/373

### CODE REVIEW

**Comment by johananl** [comment]
>I didn't touch the quickstarts yet since they're being reworked.

I'll update the quickstarts if this PR gets merged first.

File: docs/configuration-reference/backend/s3.md [hunk 1]
```diff
@@ -58,3 +55,43 @@ environment variables or in the config above.
 
 >NOTE: If no value is passed for `dynamodb_table`, installer will not use the state locking feature.
 
+## IAM permissions
+
+The following permissions are required for using the AWS S3 backend.
```
Comment (johananl) [inline docs/configuration-reference/backend/s3.md:60]:
```suggestion
The following permissions are required for using the AWS S3 backend:
```

**Comment by iaguis** [comment]
Addressed nits.

**Comment by iaguis** [comment]
I'll merge assuming @johananl is good with it after I addressed his nits.

**Pull request merged**